cloverpy/clover.py

```diff
@@ -15,15 +15,56 @@ def convert(coverage_xml):
     tree = ET.parse(coverage_xml).getroot()
     root = tree
     timestamp = root.attrib.get('timestamp')
-    files = len(root.findall('.//class'))
-    loc = len(root.findall('.//line'))
-    elements = len(filter(
-        lambda x: x.attrib.get('hits') == '1',
-        root.findall('.//line')))
+    
+    """
+    Dictionary in the format
+    {
+      (str) class_name: {
+        (str) line_number: (bool) was_line_hit
+      }
+    }
+    """
+    class_files = {}
+
+    # Iterate through all class tags in the XML (some classes may be present multiple times)
+    for class_file in root.findall('.//class'):
+        filename = class_file.get('name')
+
+        # Initialize class names that haven't been seen before
+        if not filename in class_files:
+            class_files[filename] = {}
+
+        # Store the class loc coverage dict
+        class_coverage = class_files[filename];
+
+        # Iterate over all lines of code for this instance of the class in the XML
+        for loc in class_file.findall('.//line'):
+            line_number = loc.get('number')
+
+            # Initialize lines in the code that haven't been seen before as not hit
+            if not line_number in class_coverage:
+                class_coverage[line_number] = False
+
+            # If the line has been hit, mark it as such
+            if loc.get('hits') != '0':
+                class_coverage[line_number] = True
```

Comment:
That doesn't quite work if the class is present multiple times with different coverage. If the loc for that class was ever hit once, then it should be marked as true. This would set it to `False` if the line wasn't hit in a later package.